src/lumigo_tracer/wrappers/http/http_parser.py

```diff
@@ -370,11 +440,14 @@ def get_parser(url: str, headers: Optional[dict] = None) -> Type[Parser]:  # typ
         return KinesisParser
     elif service == "events":
         return EventBridgeParser
-    elif safe_split_get(url, ".", 1) == "s3" or safe_split_get(url, ".", 0) == "s3":
+    elif safe_split_get(host, ".", 1) == "s3" or safe_split_get(host, ".", 0) == "s3":
         return S3Parser
     # SQS Legacy Endpoints: https://docs.aws.amazon.com/general/latest/gr/rande.html
-    elif service in ("sqs", "sqs-fips") or "queue.amazonaws.com" in url:
-        return SqsParser
-    elif "execute-api" in url:
+    elif service in ("sqs", "sqs-fips") or "queue.amazonaws.com" in host:
+        using_json_protocol = (
+            lowercase_headers.get("content-type", "").lower() == "application/x-amz-json-1.0"
```

Comment:
`1.0` is the version?
Do we want that this `if` will be false when they upgrade version?
I think that `SqsJsonParser` is a better fallback to `"application/x-amz-json-1.1"` than `SqsXmlParser`

Reply:
Im parsing specifically for `1.0` version because that is what the AWS documentation specifies must be in the HTTP request for SQS json request (see https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-making-api-requests-json.html#structure-post-request)

And about the default parsing, currently XML is the default protocol for communicating with SQS, and amazon wrote in their blog post that only once they see a large enough amount of users move to json that they will make it the default in their client libraries. So if most cases will use XML parsing, I want xml to be the default option (In the future if json will become the default we can make json the default)

Reply:
I meant that you should change line 448 to:
```
            lowercase_headers.get("content-type", "").lower().startswith("application/x-amz-json-")
```